src/block/stdio_executor.rs

```diff
@@ -202,28 +250,104 @@ impl<B: Backend> StdIoBackend<B> {
                 }
             }
             RequestType::Flush => return self.inner.fsync().map(|_| 0).map_err(Error::Flush),
+            RequestType::Discard | RequestType::WriteZeroes => {
+                if total_len % DiscardWriteZeroes::LEN != 0 {
+                    return Err(Error::InvalidDataLength);
+                }
+                for (data_addr, data_len) in request.data() {
+                    let mut available_bytes = *data_len as u64;
+                    let mut crt_addr = *data_addr;
+                    
+                    while available_bytes >= DiscardWriteZeroes::LEN {
```

Comment:
Until we add support to easily read data across consecutive descriptors, it looks like we might end up interpreting things a bit erroneously here if a `DiscardWriteZeros` object starts at the end of one descriptor and spills over the next. How about we replace the `if total_len % DiscardWriteZeroes::LEN != 0` block above with a similar check for every `data_len` in this `for` loop? If `data_len` is always a multiple of the object len (which AFAIK happens for all driver implementations we're aware of), then we know for sure all `DiscardWriteZeros` structs found in a descriptor are complete.

Reply:
:+1: makes sense, done